### ISSUE TITLE
Favorites not loading.

### ISSUE THREAD

**Issue by Yoposoc** [ZerataX/unlimted_favorites#3]
Hey, still alive there?

I still am hoping I can use something like this. I can't access any of the ones I favorited in the custom ones.

![Untitled](https://user-images.githubusercontent.com/2718371/69473614-45af4e80-0d6b-11ea-9583-bbf707cead4f.png)

**Comment by zeratax** [comment]
Oh yeah I actually started reworking all of this a while back. If you could export your favs so I can use them as test data that would help a lot.

**Comment by Yoposoc** [comment]
oh bless.

not sure how to export favorites but I could export settings. 
https://a.uguu.se/Qy6GJaDSuCbL_unl_fav.json

**Comment by Yoposoc** [comment]
any update to this?

**Comment by Yoposoc** [comment]
Heyo... I'm guessing you won't fix?

**Comment by Yoposoc** [comment]
So I just checked that json file and discovered there's only a couple entries under my "custom" favorite lists. One entry in "Backlog #1" and two in "Things to translate". 

Where did all of them go?
I even tried to see in the code how it fetches those numbers that I see when I navigate to my favorites page on exh. I couldn't figure it out.

At the very least is there a way to export them somehow so I can have a list of the galleries that were in a custom favorites list?

**Comment by zeratax** [comment]
oh man sorry about the extreme delay. the past few months were pretty stressful. 
So ehm hmm what do you see in the console (ctrl + shift + i). it should just print the raw saved variable to console, so either it's there or it's gone

**Comment by Yoposoc** [comment]
Ok, I turned off all other extensions and turned on unlimited_favs. I then navigated to one of the custom favlists I had. This is what it looks like:
![image](https://user-images.githubusercontent.com/2718371/83680003-2d7b8000-a595-11ea-8fe5-c9c8607efb27.png)

Here is a gif that shows this behavior. As you can see it first "goes" to my last non-custom list and then a few seconds later (it's usually within 1 second) loads up a bugged view:
![2020-06-03_16-15-14](https://user-images.githubusercontent.com/2718371/83698106-76433100-a5b5-11ea-865b-d3ff0130837d.gif)


This is the console output:
![image](https://user-images.githubusercontent.com/2718371/83679903-058c1c80-a595-11ea-8817-14013a881578.png)
The first 3 object logs (with an array of 3 elements) are shown each time it needs to load the page. Clicking the array shows properly the three favlists I'm supposed to have. 
![image](https://user-images.githubusercontent.com/2718371/83680099-569c1080-a595-11ea-9517-06def9cb6bae.png)

The array of 25 came up once I clicked on `Backlog #1` AND it bugged out to show nothing. Only after that did that console entry show up. 
Looking at one of the elements in that array, I can confirm it is a record of a gallery that I put into `backlog #1`.
![image](https://user-images.githubusercontent.com/2718371/83680787-47699280-a596-11ea-9725-45cd24257d7c.png)
![image](https://user-images.githubusercontent.com/2718371/83680810-4e90a080-a596-11ea-96ff-a5a0f2ae5d8f.png)
however, not sure why it's only 25 elements.

**Comment by Yoposoc** [comment]
here's another showing of the bugged behavior https://z.zz.ht/6dvvC.gif

**Comment by zeratax** [comment]
hmm are you using tampermonkey or greasemonkey? 
I think tampermonkey has an option to complete backup your extensions with settings and all.
trying hard to get a working version out, but I'd rather first make sure you don't lose access to your current lists

**Comment by zeratax** [comment]
oh I guess greasemonkey doesn't work with the script at all?
well anyways in the dashboard under utilities you have the option to zip export and in the zip there should be a file called `unlimited favs.storage.json` which should be the just raw json

**Comment by Yoposoc** [comment]
> hmm are you using tampermonkey or greasemonkey?

I'm using violentmonkey actually, which I believe is similar to greasemonkey.

Now that you mentioned it, I see a "values" tab in violentmonkey which has a `favsJson` that looks right! :D
https://p.teknik.io/Raw/l8p70

**Comment by zeratax** [comment]
Okay I still need to do this for the other display modes and I want to make the search work this time, but this looks pretty good imo
![grafik](https://user-images.githubusercontent.com/5024958/83938093-107fc080-a7d2-11ea-8a0f-86605fcf6bb7.png)

tags are all misc only cos I didn't use to save the category data back then, but I do now and I'll add an update function soon enough. when you add new galleries it looks fine:

![grafik](https://user-images.githubusercontent.com/5024958/83938181-9e5bab80-a7d2-11ea-97a2-ce306fa58f9e.png)

also non standard thumbnail sizes look shitty for now but w/e

**Comment by zeratax** [comment]
![update button](https://user-images.githubusercontent.com/5024958/83941304-927be380-a7ea-11ea-9b7b-1ed494945baa.png)

Alright now there's an update button in the settings that will redownload all tags and stuff from the api, which then results in the tags being properly displayed:

![updated tags](https://user-images.githubusercontent.com/5024958/83941328-b3443900-a7ea-11ea-986c-04ea0c4757b8.png)

Don't think I can do tag highlighting though, well not that easily at least :/

**Comment by Yoposoc** [comment]
> also non standard thumbnail sizes look shitty for now but w/e

yeah I think that's much lower priority. Awesome work on this, thanks a lot.

What do you mean by tag highlighting? I'm using a different userscript for that, maybe you just need to ensure no conflicts exist? I'll try it out right now.

**Comment by Yoposoc** [comment]
well uh you didn't push your updated script anywhere.... so....

**Comment by Yoposoc** [comment]
well uh you didn't push your updated script anywhere.... so....

**Comment by zeratax** [comment]
the standard tag highlighting from the website, no userscript. well also not really important.

Anyways okay I'll push an update to the other branch, so you can install it as a separate script, but do expect bugs.

install: https://github.com/ZerataX/unlimted_favorites/raw/ZerataX-cleanup/unlfavs.user.js

your `favJson` is currently in a string and not json format, so to import it you need to remove the first `"` and the last `"` and replace every instance of `\"` with `"`

also got a search working that does everything except for the `$` sign
![search](https://user-images.githubusercontent.com/5024958/83958225-c0efd200-a86f-11ea-85b2-9d8aab43846e.png)

oh also when adding favorites that currently only works by double clicking the item, not with the apply button.

**Comment by zeratax** [comment]
Hmm  just noticed that the search is kinda dumb.
if you currently search for `english p:"kantai collection"` you get the union instead of the intersection.
also currently i only use the shortcodes for namespaces e.g. `f` instead of `female`

everything else should work as expected, if not please report it

**Comment by Yoposoc** [comment]
cool, I'm testing it out and I'll create issues as needed. Found some already.

**Comment by zeratax** [comment]
had some time today and now also added thumbnail mode in 8bd75ad 
![grafik](https://user-images.githubusercontent.com/5024958/84556045-8333f300-ad20-11ea-9f40-5a59e2cf4cd2.png)

I think search now really works well, just needs a proper button and enter function I guess.

**Comment by Yoposoc** [comment]
nice!

**Comment by Yoposoc** [comment]
are you going to merge the branch so that whoever has the script will autoupdate?

**Comment by zeratax** [comment]
> are you going to merge the branch so that whoever has the script will autoupdate?

Yeah but I first want to fix a few more things.
I just finished the other favorite page display modes.
Now I just want to:
- [x] introduce a warning for a new version so that you can backup before updating
- [x] Make sure updating galleries (e.g. changing notes) doesn't update the favorited timestamp
- [x] #7 
- [x] and a proper search button

which is all really quickly done, just need to find some time

**Comment by Yoposoc** [comment]
gonna guess no progress on this?

**Comment by zeratax** [comment]
@Yoposoc fixed the last issues and made it the default version.
so hopefully it updates for everyone automatically, but I'm not sure.
it does import old saves and fixes them, tho I do recommend pressing the `update all` button once in the settings page (that gets the latest tags for every gallery and sorts them properly). 
it will take a few seconds so wait until a finished dialogue appears.

**Comment by Yoposoc** [comment]
Got it. Will report back if I have issues again.

**Comment by Yoposoc** [comment]
doesn't work again anymore. all other extensions are disabled. Been a while since I checked this so not sure when it started breaking.
